src/extension.ts

```diff
@@ -80,17 +82,55 @@ let decorationType: TextEditorDecorationType = window.createTextEditorDecoration
 
 const config = workspace.getConfiguration("metals");
 
+interface Activation {
+  shouldCreateNewProject: boolean;
+  status: Disposable;
+}
+
 export async function activate(context: ExtensionContext) {
   detectLaunchConfigurationChanges();
   checkServerVersion();
+  let activation: Activation = {
+    shouldCreateNewProject: false,
+    status: window.setStatusBarMessage("$(gear~spin) Activating Metals"),
+  };
+
+  context.subscriptions.push(
+    commands.registerCommand("metals.new-scala-project-activate", () => {
+      activation.shouldCreateNewProject = true;
+    })
+  );
 
   getJavaHome(workspace.getConfiguration("metals").get("javaHome"))
-    .then((javaHome) => fetchAndLaunchMetals(context, javaHome))
+    .then((javaHome) => fetchAndLaunchMetals(context, javaHome, activation))
     .catch((err) => {
       outputChannel.appendLine(err);
       showMissingJavaMessage();
+      activation.status.dispose();
     });
   commands.executeCommand("setContext", "metals:enabled", true);
+  setHasBuildTool();
+}
+
+function setHasBuildTool(): void {
+  let allPromises = [
+    "build.sbt",
+    " build.sc",
+    "project/build.properties",
+    "**/scala/**",
```

Comment:
Is this pattern necessary? I am concerned this glob will timeout in large workspaces causing unexpected behavior. I work on a daily basis in a git monorepo where the following command takes >1min to complete `time find . -type d -name scala | wc -l`

Reply:
It's more fool proof, but it does cause us some issues. We could maybe have a couple of more specific ones? Something like `src/main/scala/**` and  `*/src/main/scala/**` should cover more bases and be much faster.

Reply:
Otherwise, I have no idea how to support maven, pants and gradle.

Reply:
I simplified the patterns a bit, which should be much more efficient:
```
    "workspaceContains:build.sbt",
    "workspaceContains:build.sc",
    "workspaceContains:project/build.properties",
    "workspaceContains:src/main/scala",
    "workspaceContains:*/src/main/scala",
    "workspaceContains:*/*/src/main/scala"
```
This should cover most cases and in all other Metals will still activate after opening a file.